test/preprocessor/edl/preprocessor.edl

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+enclave {
+  import "ifdef_ecall.edl"
+  import "ifdef_enum.edl"
+  import "ifdef_ocall.edl"
+  import "ifdef_struct.edl"
+  import "ifdef_import.edl"
+  import "ifdef_import_from_all.edl"
+  import "ifdef_import_from.edl"
+  import "ifndef_ecall.edl"
+  import "ifndef_enum.edl"
+  import "ifndef_ocall.edl"
+  import "ifndef_struct.edl"
+  import "ifndef_import.edl"
+  import "else_ecall.edl"
+  import "else_enum.edl"
+  import "else_ocall.edl"
+  import "else_struct.edl"
+  import "else_import.edl"
```

Comment:
How do we lock down that parts of edl are not being imported?
Given
```c
#if P
void foo();
#else
void bar(); 
#endif

```

We want to make sure that either foo or bar is being imported; but not both.

Reply:
This case is implicitly tested (by the linker).  For example, in `else_ecall.edl`
```
enclave {
  trusted {
#ifdef TEST_IF_ECALL
    public int enc_ifdef_else_ecall(int magic);
#else
    public int enc_else_ecall(int magic);
#endif
  };
};
```

The enclave currently only implements `enc_else_ecall`. If both the ecall are imported, the linker will raise error as it cannot find the implementation of `enc_ifdef_else_ecall`.

Reply:
Great!